src/crate/client/tests.py

```diff
@@ -108,15 +108,22 @@ def setUpMocked(test):
 crate_port = 44209
 crate_transport_port = 44309
 local = '127.0.0.1'
-crate_layer = CrateLayer('crate',
-                         crate_home=crate_path(),
-                         port=crate_port,
-                         host=local,
-                         transport_port=crate_transport_port,
-                         settings=settings)
-
 crate_host = "{host}:{port}".format(host=local, port=crate_port)
 crate_uri = "http://%s" % crate_host
+crate_layer = None
+
+
+def ensure_cratedb_layer():
+    global crate_layer
+
+    if crate_layer is None:
+        crate_layer = CrateLayer('crate',
+                                 crate_home=crate_path(),
+                                 port=crate_port,
+                                 host=local,
+                                 transport_port=crate_transport_port,
+                                 settings=settings)
+    return crate_layer
```

Comment:
I don't really understand how this improves the situation?

Reply:
Because then, I can selectively manually disable some tests within the backplane `tests.py` and make the test suite still partially run on macOS. That is an alternative to the other patch at #386.

Based on this improvement, I might submit another patch which can enable/disable specific tests selectively by providing some environment variables.

Reply:
Just to make sure I understand this right: You want to be able to run individual tests which do not need the test layer without them triggering the test layer startup? (And the test selection is done as usual with the test runner) ?

If so :+1:  
but maybe you could add this information as comment around this code?

Reply:
I just added 0e59669c9 to describe the use case a bit more.

> You want to be able to run individual tests which do not need the test layer without them triggering the test layer startup?

Exactly.

> And the test selection is done as usual with the test runner.

I am not sure if this is possible yet, because the tests in question are doctests. I tried some variants, but I haven't been able to select specific ones based on e.g. `--tests-pattern` yet.